neqo-crypto/src/constants.rs

```diff
@@ -27,7 +27,7 @@ pub const TLS_EPOCH_APPLICATION_DATA: Epoch = 3_u16;
 macro_rules! remap_enum {
     { $t:ident: $s:ty { $( $n:ident = $v:path ),+ $(,)? } } => {
         pub type $t = $s;
-        $( pub const $n: $t = $v as $t; )+
+        $(#[allow(clippy::cast_possible_truncation)] pub const $n: $t = $v as $t; )+
```

Comment:
For my understanding: why are all these `c_uint`s (effectively Rust `u32`) mapped to smaller types, and thus risking possible cast truncations?

Reply:
I don't know. Question for @martinthomson I guss. Compatibility with NSS?

Reply:
This is a mismatch in style between Rust and C.  In NSS, we use enums (which are essentially untyped, hence `c_uint`) to define a bunch of constants.

However, the TLS protocol defines these types to have a fixed-size type, like `u16`.  If we are to present these types in our API in a way that makes sense (i.e., does not allow for values outside the permitted range) then they need to be cast down.

What I'd prefer is if we could ask the compiler to fail if a truncation occurred.  But we don't seem to have that option, or at least we didn't at the time I wrote this.